src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/LegacyTagHelperCompletionService.cs

```diff
@@ -0,0 +1,387 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Components;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    // This class is utilized entirely by the legacy Razor editor and should not be touched to avoid breaking functionality.
```

Comment:
```suggestion
    // This class is utilized entirely by the legacy Razor editor and should not be touched except when specifically working on the legacy editor to avoid breaking functionality.
```